packages/flutter_ai_providers/lib/src/providers/openwebui.dart

```diff
@@ -0,0 +1,307 @@
+  Map<String, dynamic> toJson() => {
+    'model': model,
+    'stream': true,
+    'messages': messages.map((message) => message.toOwuiJson(
+      message == messages.lastWhere((m) => m.origin == MessageOrigin.user) ? images?.map((image) => image.toJson()).toList() : null)
+    ).toList(),
+    'files': files?.map((file) => file.toJson()).toList(),
+  };
+
+  String toJsonString() => jsonEncode(toJson());
+}
+
+/// Internal open-webui json encoder / decoder
+/// Decode a chat response from the open-webui API.
+class _OwuiChatResponse {
+  final List<_OwuiChatResponseChoice> choices;
+
+  /// Creates an instance of [_OwuiChatResponse].
+  ///
+  /// [choices] is the list of choices in the response.
+  _OwuiChatResponse({required this.choices});
+
+  /// Creates an instance of [_OwuiChatResponse] from a JSON object.
+  factory _OwuiChatResponse.fromJson(Map<String, dynamic> json) {
+    return _OwuiChatResponse(
+      choices: (json['choices'] as List).map((choice) => _OwuiChatResponseChoice.fromJson(choice)).toList(),
+    );
+  }
+
+  factory _OwuiChatResponse.fromJsonString(String jsonString) {
+    return _OwuiChatResponse.fromJson(jsonDecode(jsonString));
+  }
+}
+
+/// Internal open-webui json encoder / decoder
+/// Decoder for the choice part of [_OwuiChatResponse].
+class _OwuiChatResponseChoice {
+  final ChatMessage message;
+
+  /// Creates an instance of [_OwuiChatResponseChoice].
+  ///
+  /// [message] is the message in the choice.
+  _OwuiChatResponseChoice({required this.message});
+
+  /// Creates an instance of [_OwuiChatResponseChoice] from a JSON object.
+  factory _OwuiChatResponseChoice.fromJson(Map<String, dynamic> json) {
+    return _OwuiChatResponseChoice(
+      message: _OwuiMessage.fromOwuiJson(json),
+    );
+  }
+}
+
+/// Internal open-webui json encoder / decoder
+class _OwuiImageAttachment {
+  final String type;
+  final Map<String, String> imageUrl;
+  final String name;
+
+  _OwuiImageAttachment({
+    required this.type,
+    required this.imageUrl,
+    required this.name,
+  });
+
+  Map<String, dynamic> toJson() => {
+    'type': type,
+    'image_url': imageUrl,
+    'name': name,
+  };
+
+  factory _OwuiImageAttachment.fromImageAttachment(ImageFileAttachment attachment) {
+    final base64Image = base64Encode(attachment.bytes);
+    return _OwuiImageAttachment(
+      type: 'image_url',
+      imageUrl: {'url': "data:${attachment.mimeType};base64,$base64Image"},
+      name: attachment.name,
+    );
+  }
+}
+
+/// Internal open-webui json encoder / decoder
+class _OwuiFileAttachment {
+  final String type;
+  final String id;
+
+  _OwuiFileAttachment({
+    required this.type,
+    required this.id,
+  });
+
+  Map<String, dynamic> toJson() => {
+    'type': type,
+    'id': id,
+  };
+}
+
+/// A provider for [open-webui](https://openwebui.com/)
+/// Use open-webui as unified chat provider.
+class OpenwebuiProvider extends LlmProvider with ChangeNotifier {
```

Comment:
I would rename it to `OpenWebUIProvider`

Reply:
Done